spec.md

```diff
@@ -5,4599 +5,421 @@ keywords: registry, on-prem, images, tags, repository, distribution, api, advanc
-An untrusted registry returns a manifest.
-Before proceeding to download the individual layers, the engine verifies the manifest's signature, ensuring that the content was produced from a trusted source and no tampering has occurred.
-After each layer is downloaded, the engine verifies the digest of the layer, ensuring that the content matches that specified by the manifest.
+1. **Pull** - Clients are able to pull from the registry
+2. **Push** (OPTIONAL) - Clients are able to push to the registry
+3. **Content Discovery** (OPTIONAL) - Clients are able to list or otherwise query the content stored in the registry
+4. **Content Management** (OPTIONAL) - Clients are able to control the full life-cycle of the content stored in the registry
 
-### Resumable Push
+In order to test a registry's conformance against these workflow categories, please use the [conformance testing tool](./conformance/).
 
-Company X's build servers lose connectivity to a distribution endpoint before completing an image layer transfer.
-After connectivity returns, the build server attempts to re-upload the image.
-The registry notifies the build server that the upload has already been partially attempted.
-The build server responds by only sending the remaining data to complete the image file.
+### Official Certification
 
-### Resumable Pull
+Registry providers can self-certify by submitting conformance results to [opencontainers/oci-conformance](https://github.com/opencontainers/oci-conformance).
 
-Company X is having more connectivity problems but this time in their deployment datacenter.
-When downloading an image, the connection is interrupted before completion.
-The client keeps the partial data and uses http `Range` requests to avoid downloading repeated data.
+### Workflow Categories
 
-### Layer Upload De-duplication
+#### Pull
 
-Company Y's build system creates two identical layers from build processes A and B.
-Build process A completes uploading the layer before B.
-When process B attempts to upload the layer, the registry indicates that its not necessary because the layer is already known.
+##### Pulling Blobs
 
-If process A and B upload the same layer at the same time, both operations will proceed and the first to complete will be stored in the registry (Note: we MAY modify this to prevent dogpile with some locking mechanism).
+To pull a blob, perform a `GET` request to a url in the following form:
+[2a](#Endpoints) `/v2/<name>/blobs/<digest>`
 
-## Changes
+`<name>` is the namespace of the repository, and `<digest>` is the blob's digest.
 
-The V2 specification has been written to work as a living document, specifying only what is certain and leaving what is not specified open or to future changes.
-Only non-conflicting additions SHOULD be made to the API and accepted changes SHOULD avoid preventing future changes from happening.
+A GET request to an existing blob URL MUST provide the expected blob, with a reponse code that MUST be `200 OK`.
```

Comment:
Ignorant question - is there any question of permissions here?

Reply:
We are avoiding discussion of the authentication/authorization system. There is a "Docker-style" system, but it is not under the purview of this spec. The response codes assume successful authorization.